backend/src/seeds/factory/transactionLink.ts

```diff
@@ -0,0 +1,43 @@
+import { ApolloServerTestClient } from 'apollo-server-testing'
+import { createTransactionLink } from '@/seeds/graphql/mutations'
+import { login } from '@/seeds/graphql/queries'
+import { TransactionLinkInterface } from '@/seeds/transactionLink/TransactionLinkInterface'
+import { transactionLinkExpireDate } from '@/graphql/resolver/TransactionLinkResolver'
+import { TransactionLink } from '@entity/TransactionLink'
+import { User } from '@entity/User'
+
+export const transactionLinkFactory = async (
+  client: ApolloServerTestClient,
+  transactionLink: TransactionLinkInterface,
+): Promise<void> => {
+  const { mutate, query } = client
+
+  // login
+  await query({ query: login, variables: { email: transactionLink.email, password: 'Aa12345_' } })
+
+  const variables = {
+    amount: transactionLink.amount,
+    memo: transactionLink.memo,
+  }
+
+  await mutate({ mutation: createTransactionLink, variables })
+
+  if (transactionLink.createdAt || transactionLink.deletedAt) {
+    const user = await User.findOneOrFail({ where: { email: transactionLink.email } })
+    const dbTransactionLink = await TransactionLink.findOneOrFail({
```

Comment:
instead of querying the link and guessing which one it is, just use the value returned by the `mutate` creating it.

```
async createTransactionLink(
    @Args() { amount, memo }: TransactionLinkArgs,
    @Ctx() context: any,
  ): Promise<TransactionLink>
```

Reply:
Same applies to the user in the last PR:

```
async login(
    @Args() { email, password, publisherId }: UnsecureLoginArgs,
    @Ctx() context: any,
  ): Promise<User>
```

Reply:
Yeah! Very good idea. Therefore I refactored the `createUser` mutation, which now return a user object too. Therefore the mutation in the frontend must be adopted.